cmd/codegen/app/generate_openapi.go

```diff
@@ -152,13 +152,20 @@ func (o *CreateClientOpenAPIOptions) Run() error {
 		}
 	}
 
-	defer cleanupFunc()
-
 	err = generator.InstallOpenApiGen(o.GeneratorVersion, gopath)
+	// Reset the go.mod and go.sum after installing openapi-gen
+	cleanupFunc()
 	if err != nil {
 		return errors.Wrapf(err, "error installing kubernetes openapi tools")
 	}
 
+	cleanupFunc, err = util.BackupGoModAndGoSum()
```

Comment:
> This ends up requiring some trickery with `go.mod` - installing
`openapi-gen` ends up removing `k8s.io/api` and `k8s.io/apimachinery`
from `go.mod`, resulting in the openapi being generated against master
of those packages rather than the version we're actually depending on,
so we need to copy/restore `go.mod` and `go.sum` twice - before/after
installing `openapi-gen`, and before/after actually generating the openapi.

oh joy